providers/base/bin/ishtp.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+import subprocess
+import sys
+import os
+import argparse
+
+
+def get_release_version():
+    try:
+        output = subprocess.check_output(
+            ["lsb_release", "-rs"], text=True
+        ).strip()
+        return int(float(output))
+    except (subprocess.CalledProcessError, ValueError):
+        print("Error: Unable to determine release version.")
+        sys.exit(1)
+
+
+def is_module_loaded(module_name):
+    try:
+        output = subprocess.check_output(["lsmod"], text=True)
+        return module_name in output
+    except subprocess.CalledProcessError:
+        return False
+
+
+def check_modules():
+    release = get_release_version()
+
+    if release >= 24:
+        expected_modules = ["intel_ishtp_hid", "intel_ish_ipc", "intel_ishtp"]
+    else:
+        expected_modules = [
+            "intel_ishtp_loader",
+            "intel_ishtp_hid",
+            "intel_ish_ipc",
+            "intel_ishtp",
+        ]
+
+    exit_code = 0
+    for module in expected_modules:
+        print("Checking module: {}".format(module))
+        if not is_module_loaded(module):
+            print("FAIL: The '{}' module is not loaded!".format(module))
+            exit_code = 1
+        else:
+            print("PASS: It's loaded")
+        print()
+
+    return exit_code
```

Comment:
There is not need to handle all of these exit codes. Just raise an error if a module could not be detected
```suggestion
    modules_list = get_module_list()
    missing_modules = [
        module for module in expected_modules if module not in modules_list
    ]
    if missing_modules:
        print("Missing ISHTP kernel modules:")
        for module in missing_modules:
            print(" - {}".format(module))
        raise SystemExit("Missing some ISHTP modules")
```